### ISSUE TITLE
Invalid Spec: unknown option: rw

### ISSUE THREAD

**Issue by saamalik** [docker/docker#32594]
**Description**

The latest _Linux_ docker-engine doesn't like the volume specification generated by docker-compose config. As a simple example, starting with this docker-compose file:
```
version: "3"

services:
  mongodb:
    image: mongo:3.4
    volumes:
        - mongo_volume:/data/db
    networks:
      - msc
    ports:
      - 27017:27017
volumes:
    mongo_volume:
networks:
   test:
```

Generate a docker-compose config (needed to collapse compose extension files):
```
docker-compose -f docker-test.yaml config > docker-ready.yaml
```

This results in the following file:
```
networks:
  test: {}
services:
  mongodb:
    image: mongo:3.4
    networks:
      msc: null
    ports:
    - 27017:27017/tcp
    volumes:
    - mongo_volume:/data/db:rw
version: '3.0'
volumes:
  mongo_volume: {}
```
So far so good, however if I try to deploy the docker-ready.yaml, e.g:
```
[vagrant@msc-vbox ~]$ docker stack deploy -c docker-ready.yaml test1
1 error(s) decoding:

* invalid spec: mongo_volume:/data/db:rw: unknown option: rw
```
Removing the _:rw_ from the docker-ready.yaml fixes the issue, but this config used to work perfectly fine. I believe it has something to do with the updated docker-engine on Linux (CentOS 7).

However, what's interesting is that deploying the same generated docker-ready.yaml on Docker For Mac (17.0.3) works _great_. Looks like a regression in only Linux Docker.

**Output of `docker version`:**

```
Docker version 17.04.0-ce, build 4845c56
```

**Output of `docker info`:**

```
Containers: 8
 Running: 8
 Paused: 0
 Stopped: 0
Images: 8
Server Version: 17.04.0-ce
Storage Driver: overlay
 Backing Filesystem: xfs
 Supports d_type: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: gbmjruhn5q5l3cq62m0balt35
 Is Manager: true
 ClusterID: jy91b9y3hh9rwa1swx091gsjz
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.0.2.15
 Manager Addresses:
  10.0.2.15:2377
Runtimes: runc
Default Runtime: runc
Init Binary:
containerd version: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.10.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.702GiB
Name: msc-vbox
ID: MSGX:2BI6:B7FN:LYC2:GIG5:DARL:5VXZ:3QHO:TTUB:AYSB:WONY:EO6W
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
CentOS 7 (VirtualBox)

**Comment by mkumatag** [comment]
seems like dup of https://github.com/docker/docker/issues/32398 and got fixed part of https://github.com/docker/docker/pull/32407.

**Comment by cpuguy83** [comment]
Yep.
This is fixed in 17.05.

Thanks!
